### PR TITLE
fix: grpc-transcode plugin: fix map data population

### LINKED ISSUE #8655: bug: GRPC transcode can't work with protobuf.Struct
### Current Behavior

# Problem
The inner struct object has not been parsed correctly. and is empty when using grpc-transcode plugin.
# details
I have a grpc serivce with has below definition:
```
--- the request part
message RawRequest {
  google.protobuf.Struct reqdata = 1;
  string type = 2;
  string serviceurl = 3;
  string source = 4;
  int32 page = 5;
  int32 size = 6;
}
```
and I try to send a post request in postman with below request args:
```
{
    "type":"7",
    "serviceurl" :"xxxx",
    "reqdata" :{
        "key":"yyyy",
        "key_type":"zzzz",
        "model":"tttt"
    },
    "source":"3"
}
```
and When I checking the logs, I found the reqdata object is empty (not null). 

### Expected Behavior

The struct object should has data.

### Error Logs

_No response_

### Steps to Reproduce

1. Create a routing from rest to grpc with grpc transcode.
2. and checking whether the grpc service side can receive the correct data.

### Environment

apisix:3.0.0

### PR DESCRIPTION
**Pull request by kingluo**
### Description

Fixes #8655

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: t/grpc_server_example/go.mod [hunk 1]
```diff
@@ -3,6 +3,7 @@ module github.com/api7/grpc_server_example
 go 1.11
 
 require (
+	github.com/golang/protobuf v1.5.0 // indirect
```
Comment (spacewander) [inline t/grpc_server_example/go.mod:6]: Please run ` go mod tidy` to remove indirect mark

File: apisix/plugins/grpc-transcode/util.lua [hunk 2]
```diff
@@ -164,15 +165,33 @@ function _M.map_message(field, default_values, request_table)
...
+                if ty == "map" then
+                    local tbl
+                    for k, v in pairs(request_table[name]) do
+                        tbl, err = _M.map_message(field_type,
+                            default_values and default_values[name],
+                            request_table[name], k)
+                        if not sub then
+                            sub = {}
+                        end
+                        sub[k]= tbl[k]
```
Comment (spacewander) [inline apisix/plugins/grpc-transcode/util.lua:177]: Let's check err before accessing tbl

Comment (spacewander) [inline apisix/plugins/grpc-transcode/util.lua:177]:
```suggestion
                        sub[k] = tbl[k]
```

File: t/plugin/grpc-transcode-struct.t [hunk 3]
```diff
@@ -0,0 +1,121 @@
+#
```
Comment (spacewander) [inline t/plugin/grpc-transcode-struct.t:1]: Could we move these tests under https://github.com/apache/apisix/blob/master/t/plugin/t/plugin/grpc-transcode3.t? This file is small and doesn't have special setup.

File: t/plugin/grpc-transcode-struct.t [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin")
+            local json = require("toolkit.json")
+
+            local content = t.read_file("t/grpc_server_example/echo.pb")
+            local data = {content = ngx.encode_base64(content)}
+            local code, body = t.test('/apisix/admin/protos/1',
+                 ngx.HTTP_PUT,
```
Comment (spacewander) [inline t/plugin/grpc-transcode-struct.t:45]:
```suggestion
                ngx.HTTP_PUT,
```

**Review by spacewander — APPROVED** [review]

**Review by soulbird — APPROVED** [review]

**Pull request merged**